samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -128,10 +128,11 @@ public void writeContainerToHostMapping(String containerId, String hostName) {
 
   public void close() {
     metadataStore.close();
-    taskAssignmentManager.close();
   }
 
   public TaskAssignmentManager getTaskAssignmentManager() {
```

Comment:
Why make this change? It's weird for a getter to return new objects. It's also weird that init is called here but caller is responsible for calling close.

Reply:
Before using the operations from metadatastore, one has to invoke metadatastore.init(). That wasn't done, hence we'd the NPE bug referenced by @xinyuiscool in the PR. 

As we discussed offline, deleted the `getTaskAssignmentManager` method from `LocalityManager`. I agree that having a getAPI that creates objects like factory does not make much sense. In my initial migration to `MetadataStore`, we missed this part somehow. For now, deleted this method and in couple of places where this was used, simply in-lined the invocation(Other changes in the patch are mocking changes in tests).